student.rb

```diff
@@ -1,9 +1,13 @@
 require_relative './person'
+require_relative './classroom'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission: parent_permission)
     @classroom = classroom
+    classroom.add_student(self) unless classroom.students.include?(self)
```

Comment:
- [ ] Also you did pass the line number 10 without setting it on a definition of `classroom` to see if there is already someone else in that classroom who has the same name as them, it will add them to the list of students instead of adding themselves into it.

Reply:
I need to add the prevent duplicated in Classroom, but I'll keep the logic since there could be more than one student with the same name.